### PR TITLE
Fix identity provider login page not found due to missing permission

### LINKED ISSUE #4794: OAuth IDP Login Page returns a Not Found Page
Type: **bug|regression**

## Issue Description
OAuth IDP login pages created by the Hydra plugin are not showing up in the latest release candidate 6. So `/account/login` returns Not Found.

Going back a few commits (e.g `d37e12b2ea92`), the page shows up fine.

## Steps to Reproduce

1. Run the Starterkit latest, Reaction rc-6, and Hydra master
2. Try to login (or create account) on the Starterkit
3. You'll be shown a "Not Found" page on the Meteor app instead of Login form

## Possible Solution
My guess is that a new update to the Router is making it not find the needed template/component. (or maybe caused by the latest released patches?)

### Versions
Node: 8.11.3
NPM: 6.3.0
Meteor Node: 8.11.4
Meteor NPM: 6.4.1
Reaction CLI: 0.29.0
Reaction: 2.0.0-rc.6
Reaction branch: release-2.0.0-rc.6

### PR DESCRIPTION
**Pull request by impactmass**
Resolves #4794  
Impact: **major**  
Type: **bugfix**

## Issue
OAuth IDP login pages created by the Hydra plugin are not showing up after recent  updates. So `/account/login` returns Not Found.

This started happening after the permission update to Package routes publication.

## Solution
Add the permission for the Hydra plugin to guest groups. This allows the route to be published for anyone visitor to the meteor app trying to login from the Starterkit. 

This should be done for plugins that add public routes.

## Breaking changes
N/A

No migration needed.

## Testing
1. Start up Starterkit (develop), Hydra (master) and Reaction API (this branch)
2. On the Starterkit, try to login (i.e click the login button)
3. You should be redirected to the Meteor app; and you should be able to see the login form (that's the fix).
4. Test this in incognito as well.

## Docs
I'll go through the docs and creating a public that adds a public route. It should say that the permission for the plugin needs to be granted to anonymous users, otherwise the route will not be accessible to non-admin users.

**Update:** In fact, there is no docs concern. Because in a post 2.0 world, our process now is that UI should be driven separately from the API (as seen in the warning message [here](https://docs.reactioncommerce.com/docs/1.16.0/swag-shop-initialization#getting-started-creating-a-plugin)

### CODE REVIEW

**Comment by impactmass** [comment]
@mikemurray here's a reminder to check this PR.

**Comment by impactmass** [comment]
@mikemurray your finding about getting the not-found page after a reset is true. I can confirm it. It shows a fault in the way startup events happen when a reset happens. 

I think this needs to be sorted out separately. I don't know if it's a recent regression. 

The afterCoreInit is the specified way all plugins add new roles to groups at this point, so it's likely affecting others too.

**Comment by mikemurray** [comment]
@impactmass If `afterCoreInit` is having issues then this PR should be blocked by that issue since it's it would require someone to start reaction twice before it "works".

Also, the permission issue can still be overcome by going into the Accounts panel and enabling OAuth login for both "Customer" and "Guest".

**Comment by pmn4** [comment]
Hey y'all, what is the status of this?
I recently started developing on RC Platform (using the `master` branch) and am blocked by this issue. I have been forced to work only on publicly facing features (since I cannot even log in as the owner to add products or update settings)

thanks!

**Comment by spencern** [comment]
@pmn4 @mikemurray @impactmass 
I see the `blocked` label has been added here, can we connect the issue that's blocking this to the ticket? Is anyone actively working on the blocking issue right now?

**Comment by mikemurray** [comment]
@spencern it seems like this PR https://github.com/reactioncommerce/reaction/pull/4795 is attempting to address the issue of why `afterCoreInit` in this PR seems to take 2 runs to take hold.

TL;DR it seems to be migration 5

**Comment by zenweasel** [comment]
@mikemurray I'm pretty sure that Pat is trying to migrate his site to 2.0/starter-kit

**Comment by spencern** [comment]
@impactmass I don't remember merging this though GitHub says I did. I merged #4835  - should this be reverted?

**Comment by mikemurray** [comment]
@spencern I think it might not be a big deal since it's adding `reaction-hydra-oauth` permission which is the parent permission of `account/login` that was added in PR #4835.

**Comment by impactmass** [comment]
I think GitHub closed this PR because the other PR you merged was branched off the branch here. I've checked rc-7 branch and the changes there looked good.

**Comment by janus-reith** [comment]
Im experiencing the same issue on a fresh checkout of reacttion-platform, with 2.0.0-rc.7 pulled.

Only change I made is modifying the docker-compose file to make sure that the volume for the mongo container is unique.

**Comment by impactmass** [comment]
Hi @janus-reith, this fix did not go out with the [rc-7 release](@janus-reith). This fix was merged into the rc-7 **branch** shortly after we released rc-7 as a security patch. 

It will be published with the rc-8 release. If you want to test it out right now you can use the [rc-8 branch](https://github.com/reactioncommerce/reaction/tree/release-2.0.0-rc.8).

**Comment by spencern** [comment]
The merged to `release-2.0.0-rc.7` is misleading here because we had a planned release for rc.7 and ended up releasing rc.7 as a security patch and changing the planned release to rc.8
